docs/tutorials/creating-a-runtime-module.md

```diff
@@ -51,184 +73,286 @@ You should be able to successfully compile this template module with:
 cargo build
```

Comment:
Should we also use `--release` here for consistency?

Reply:
When I originally wrote this I didn't use `--release` because it takes longer and the build will never actually be executed. The module will be rebuilt when the entire node is built. Since writing this, I learned that `cargo check` is even faster than `cargo build` because it does all the syntax, type, and borrow checking but doesn't actually build an executable. I'd suggest we use `cargo check` here.

Reply:
I am pretty happy with `cargo check`

I think, for docs purposes, if we ever do `cargo build` it should include `--release`, even if it is a little slower.

If they know rust well enough, they can omit the `--release` part, if they don't know rust well enough, better not have them later think that `cargo build` is okay for running a node.

Reply:
why `cargo build` is not okay for running a node? My understanding is we get a "development version" of node, maybe more tracing, more verbose error/warning msg, less optimized memory usage, etc.

Reply:
@jimmychu0807 yes, and those things can make it so that block production is stalled.

For example, I don't think you can upload smart contracts using a non-release build of a Substrate node. The transaction will always take longer than the 6-10 second blocktime, so then a block is not produced.